src/cpr_data_access/models.py

```diff
@@ -1127,15 +1133,26 @@ def _from_huggingface_parquet(
 
         hf_dataframe: pd.DataFrame = huggingface_dataset.to_pandas()
 
+        # This undoes the renaming of columns done in to_huggingface()
+        hf_dataframe = hf_dataframe.rename(columns={"document_languages": "languages"})
+
+        # Create a dummy variable to group on combining document_id and translated.
+        # This way we get an accurate count in the progress bar.
+        hf_dataframe["_document_id_translated"] = hf_dataframe[
+            "document_id"
+        ] + hf_dataframe["translated"].astype(str)
+
```

Comment:
Nice one! What it tells me, is that `document_id` is not appropriate to be used as the unique id anymore, and this is a good workaround for that. I've faced this issue previously with the metrics CLI where upon grouping on the `document_id` the translated and non-translated results got mashed together.  So probably there is scope for this to be used more widely as part of another piece of work.

Reply:
yes – i'm hoping this will be dealt with differently in vespa (@harrisonpim)? in the first version of the pipeline that me and joel designed the json files were named `{document_id}_translated_{target_language}`